CDCgov/prime-reportstream
# 2270 Allow mappers to run regardless of field value

This PR adds a configuration parameter to the schemas to allow for a mapper to run regardless if the field has a value or not.  Previously mappers could only run if no value was present.  These changes also migrate the logic to decide when to use mappers and default values to the Element class as that logic was present in three different places.  Note the Translator logic to use a mapper is different as it looks for input values in both input and output reports.

When using the new property, the YAML will look as follows:
```
- name: abnormal_flag
    type: CODE
    valueSet: hl70078
    natFlatFileField: Abnormal_flag
    hhsGuidanceField:
    hl7Field: OBX-8
    mapper: obx8(test_result)
    mapperOverrideValues: true
```

Test Steps:
1. Run the unit and integration (`gradle test testintegration`) tests and verify that there are no errors.

## Changes
- Added mapperAlwaysRun parameter to schema config
- Moved logic on when to use mappers and default to Element class.
- Made some minor cleanups

## Checklist

### Testing

### Process

## Fixes
- #2270

## To Be Done
